src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -0,0 +1,252 @@
+                // Get current robots velocity
+                if (frame != 0 && delta_time != 0)
+                {
+                    float prev_x_pos = prev_x_pos_arr[robot_id];
+                    float prev_y_pos = prev_y_pos_arr[robot_id];
+
+                    float curr_x_pos = curr_robot_pos.getX();
+                    float curr_y_pos = curr_robot_pos.getY();
+
+                    velocity_x = (curr_x_pos - prev_x_pos) / delta_time;
+                    velocity_y = (curr_y_pos - prev_y_pos) / delta_time;
+                    speed      = std::pow(
+                        std::pow(velocity_x, 2.f) + std::pow(velocity_y, 2.f), 0.5f);
+
+                    // update previous robot x and y position
+                    prev_x_pos_arr[robot_id] = curr_x_pos;
+                    prev_y_pos_arr[robot_id] = curr_y_pos;
+                }
+                output_file << frame << "," << time << ","
+                            << std::to_string(computationTime.count()) << "," << robot_id
+                            << "," << robot_radius[robot_id] << ","
+                            << curr_robot_pos.getX() << "," << curr_robot_pos.getY()
+                            << "," << velocity_x << "," << velocity_y << "," << speed
+                            << "," << has_collided << ","
+                            << simulator.getAgentPrefVelocity(robot_id).getX() << ","
+                            << simulator.getAgentPrefVelocity(robot_id).getY()
+                            << std::endl;
+            }
+            frame++;
+            prev_frame_time = time;
+            simulator.doStep();
+            auto finishTickTime = std::chrono::high_resolution_clock::now();
+            computationTime += finishTickTime - startTickTime;
+        } while (!simulator.haveReachedGoals() && prev_frame_time < 15.f);  //);
+        auto finishTime = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double> totalTime = finishTime - startTime;
+        std::cout << "Total run time = " << totalTime.count() << std::endl;
+        std::cout << "Total simulation time = " << prev_frame_time << std::endl;
+        std::cout << "Average time per tick = " << totalTime.count() / frame << std::endl;
+
+        output_file.close();
+    }
+};
+
+TEST_F(HRVOTest, div_b_edge_test)
+{
+    const Vector2 goal_offset  = Vector2(8.f, 0);
+    const Vector2 robot_offset = Vector2(0.f, -ROBOT_RADIUS * 2.5f);
+    for (std::size_t i = 0; i < 1; ++i)
+    {
+        const Vector2 position = -1 * (goal_offset / 2) + Vector2(0.f, 2.5f) +
+                                 (static_cast<float>(i) * robot_offset);
+        simulator.addAgent(position, simulator.addGoal(position + goal_offset));
+        // const Vector2 position2 = (goal_offset / 2) + Vector2(0.f, 2.2f) +
+        // (static_cast<float>(i) * robot_offset); simulator.addAgent(position2,
+        // simulator.addGoal(position2 - goal_offset));
```

Comment:
commented out code

Reply:
Good catch, removed the commented out code